### PR TITLE
Update cluster-api to v1.2.3

### LINKED ISSUE #911: Use cluster-api 1.2.3
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.2.3 released, lets use this version

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/issues/911#tasklist-block-820a8913-029c-472f-a49e-5337d8f20898

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

**What this PR does / why we need it**:
- Update cluster-api to v1.2.3

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #911 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Update cluster-api to v1.2.3
```

### CODE REVIEW

**Review by Amulyam24 — APPROVED** [review]
/lgtm

File: hack/tools/go.mod [hunk 1]
```diff
@@ -10,10 +10,10 @@ require (
 	github.com/joelanford/go-apidiff v0.4.0
 	github.com/onsi/ginkgo v1.16.5
 	gotest.tools/gotestsum v1.8.2
-	k8s.io/code-generator v0.24.4
-	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20220912130527-3c462340b297
+	k8s.io/code-generator v0.25.0
```
Comment (mkumatag) [inline hack/tools/go.mod:13]: May I know why are we bumping this version?

File: hack/tools/go.mod [hunk 2]
```diff
@@ -10,10 +10,10 @@ require (
...
+	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20221008002318-c9b4a0f0fc38
 	sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20211110210527-619e6b92dab9
-	sigs.k8s.io/controller-tools v0.9.2
+	sigs.k8s.io/controller-tools v0.10.0
```
Comment (mkumatag) [inline hack/tools/go.mod:16]: and why this one too?

File: hack/tools/go.mod [hunk 3]
```diff
@@ -10,10 +10,10 @@ require (
 	github.com/joelanford/go-apidiff v0.4.0
 	github.com/onsi/ginkgo v1.16.5
 	gotest.tools/gotestsum v1.8.2
-	k8s.io/code-generator v0.24.4
-	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20220912130527-3c462340b297
+	k8s.io/code-generator v0.25.0
+	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20221008002318-c9b4a0f0fc38
```
Comment (mkumatag) [inline hack/tools/go.mod:14]: And hope this one pointing to `v1.2.x` branch

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: hack/tools/go.mod (hunk 1 above)
Comment (Prajyot-Parab) [reply hack/tools/go.mod:13]: auto-updated via `sigs.k8s.io/cluster-api/hack/tools` upgrade

File: hack/tools/go.mod (hunk 2 above)
Comment (Prajyot-Parab) [reply hack/tools/go.mod:16]: auto-updated via `sigs.k8s.io/cluster-api/hack/tools` upgrade

File: hack/tools/go.mod (hunk 3 above)
Comment (Prajyot-Parab) [reply hack/tools/go.mod:14]: Had updated to use latest version, verifying same.

Comment (Prajyot-Parab) [reply hack/tools/go.mod]: Fixed.

File: hack/tools/go.mod (hunk 1 above)
Comment (Prajyot-Parab) [reply hack/tools/go.mod]: Fixed.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**